dna/__main__.py

```diff
@@ -162,6 +165,32 @@ def __init__(
         self.ootsp_test_predict_time = ootsp_test_predict_time
         self.ootsp_test_scores = ootsp_test_scores
 
+    def __str__(self):
+        results = copy.deepcopy(self.__dict__)
```

Comment:
Instead of starting with everything and then removing the things we don't want, can you just start with an empy dict and build it up with the things we do want? This will save us an expensive copy.